### ISSUE TITLE
URLs whose paths begin with `/data/` do not work

### ISSUE THREAD

**Issue by eecavanna** [microbiomedata/nmdc-server#1094]
When I visit https://data.microbiomedata.org/data/nmdc:omprc-11-897qak81/nmdc:wfmgan-11-0w0rnd53.1/nmdc_wfmgan-11-0w0rnd53.1_scaffold_lineage.tsv, I expect my web browser to download a TSV file. Instead, my web browser "loads the URL" for 1.7 minutes, and then shows a Cloudflare error page saying that a timeout has occurred.

---

I looked into this and confirmed that the `frontend` deployment's logs _do_ show the incoming request (with a response status code of HTTP 499).

<img width="1056" alt="image" src="https://github.com/microbiomedata/nmdc-server/assets/134325062/fa1f97d4-6186-4c6c-8f3e-779393a295a2">

I also confirmed that the frontend code still contains a Nginx config snippet that redirects such requests to `http://data:8080`.

https://github.com/microbiomedata/nmdc-server/blob/4d366a62895d7afbdc29e69c0d0f45f00f7765ac/web/nginx.conf.template#L35-L38

On Spin, from the "perspective" of the `frontend` deployment, `http://data:8080` refers to port `8080` of a Kubernetes service named `data` in the same namespace (`nmdc`, in this case). The `data` service is a headless service that forwards all ports to a deployment named `data`, which is the data proxy deployment running the container image `nginxinc/nginx-unprivileged:stable`.

Note: When I look at the logs of the data proxy deployment, I do _not_ see any HTTP requests. I don't know whether that's normal or indicative of a problem.

More details are on Slack, at: https://nmdc-group.slack.com/archives/C01034ES34G/p1701910352685289

**Comment by shreddd** [comment]
Can we try restarting the stack? I think this may be where the proxy gets stale because the backend container was moved.

**Comment by eecavanna** [comment]
@shreddd [restarted the stack](https://nmdc-group.slack.com/archives/C01034ES34G/p1701973262439159) today (in the production environment).

Afterward, the [previously-not-working URL](https://data.microbiomedata.org/data/nmdc:omprc-11-897qak81/nmdc:wfmgan-11-0w0rnd53.1/nmdc_wfmgan-11-0w0rnd53.1_scaffold_lineage.tsv) (which is shown in the Issue description) began working again. It continues to work.

**Comment by eecavanna** [comment]
@shreddd implemented a fix in the development environment today. We are planning to implement the same fix in the production environment next week. 

In short, the fix is to reconfigure the Ingress (which currently routes everything to the frontend service) to route `/data` requests to the data proxy service, and everything else to the frontend service. In detail, the fix is to—on the Ingress—for each Rule (i.e. host), modify the existing Path so its string is `/(.*)`, create a new Path having the string `/data/(.*)`, and add an Annotation (key: `nginx.ingress.kubernetes.io/rewrite-target`, value: `/$1`).

**Comment by ssarrafan** [comment]
Appears to be in progress.  Moving to the next sprint. @eecavanna let me know if this should be in the backlog instead.

**Comment by eecavanna** [comment]
This issue has been fixed.

We fixed it by configuring the Kubernetes Ingress that sits in front of the frontend workload, to perform the path-based routing; instead of relying on the Nginx server running within the frontend workload to do it.

Here's a screenshot showing what we did in the dev environment. We did something analogous in the production environment.

![image](https://github.com/microbiomedata/nmdc-server/assets/134325062/7d0d19b0-c159-4f00-8b65-7c75165907f2)
